src/controllers/controller.h

```diff
@@ -9,6 +9,21 @@
 class ControllerJSProxy;
 class ControllerScriptEngineLegacy;
 
+enum class PhysicalTransportProtocol : uint8_t {
+    UNKNOWN,
+    USB,
+    BlueTooth,
+    I2C,
+    SPI,
+    FireWire // IEEE 1394
+};
```

Comment:
in the later commits, you're often resolving this to strings via a bunch of different techniques (switch-case or temporary map). May suggest you unify these different occurences?